### ISSUE TITLE
Integrate endpoint submission form with Jira

### ISSUE THREAD

**Issue by CharliePatterson** [digital-land/submit#615]
## Context:

- LPAs have to submit their dataset endpoints to the platform by sending an email which opens a ticket in Jira.
- We've built a form to enable the collection of the required information in a more structured format, to ensure that all the relevant information is collected.
- This form needs to be able to send the data to Jira and open a ticket. Leveraging the Jira API is the best way to get this directly into Jira in the right format.
- Owen should be able to issue API keys and secrets for both the sandbox and production JIRA Service Desks.
- We will need to update the privacy notice.
- Cyber security have defined the following set of conditions:

**Conditions:**

- Ensure that the authentication to the JIRA API is appropriately configured and not exposed
- Ensure that the authentication method to the JSM uses a strong authentication method
- Ensure that the webform captures necessary data
- Ensure that appropriate privacy policy information is available to the end user
- Ensure that the JIRA Service Management API is not publicly exposed unintentionally
- Once testing and configuration for the integration is completed, the team should consider engaging with Cyber to arrange an up to date ITHC

## Acceptance criteria:

- As a data provider, when I complete the endpoint submission form, the information I have entered into the form is sent to the relevant Jira project
- As a member of the data management team, when I open a ticket in Jira which has been created by the endpoint submission form, I see all the relevant information inputted in the correct format

## Spike document
https://docs.google.com/document/d/1tECxTpy2twzE2ZvKOCtrry4iym4vtqX67xwugeaEsQI/edit?tab=t.0

**Comment by rosado** [comment]
Is the mentioned form accessible anywhere?

**Comment by CharliePatterson** [comment]
Spike: https://github.com/digital-land/submit/issues/281

**Comment by DilwoarH** [comment]
@GeorgeGoodall-GovUk - can we get access to this: https://docs.google.com/document/d/1tECxTpy2twzE2ZvKOCtrry4iym4vtqX67xwugeaEsQI/edit?tab=t.0

**Comment by DilwoarH** [comment]
@CharliePatterson - I've added the code that calls the JIRA APIs and creates the service request.

Regarding the form, are these the views you want:

![Image](https://github.com/user-attachments/assets/a7e6f629-f2ef-429a-b6e6-3ed934aa1dc2)
![Image](https://github.com/user-attachments/assets/f84470bc-b760-4c27-ac0a-04ce2db7bcad)

A few potential UX flow issues:
- If they want to edit the organisation, the only way to do this is by returning to the organisation page.
- If they want to change the dataset, it will take them back to the start of the service
- Are all fields mandatory? What if they don't have documentation?

**Comment by alextea** [comment]
>     * If they want to edit the organisation, the only way to do this is by returning to the organisation page.
> 
>     * If they want to change the dataset, it will take them back to the start of the service

You're right — originally the submit form wasn't integrated into the dashboard as the designs pre-date it. I'll share a new flow in this thread.

>     * Are all fields mandatory? What if they don't have documentation?

Having a documentation URL is a requirement as it's how we verify that a dataset is authoritive. They shouldn't be able to publish without one — we should get confirmation on this from Swati

**Comment by DilwoarH** [comment]
Blocked until approach has been reviewed by @eveleighoj

**Comment by neilfwar** [comment]
> Blocked until approach has been reviewed by [@eveleighoj](https://github.com/eveleighoj)

@DilwoarH Is it only Owen who can do this ? he is on leave for 2 weeks & not back until 10 Feb

**Comment by DilwoarH** [comment]
> > Blocked until approach has been reviewed by [@eveleighoj](https://github.com/eveleighoj)
> 
> [@DilwoarH](https://github.com/DilwoarH) Is it only Owen who can do this ? he is on leave for 2 weeks & not back until 10 Feb

@CharliePatterson is checking with the team to see if there is anyone else who can sign off
